test/README_unittests.cpp

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: 2024 Francesco Cavaliere <francescocava95@gmail.com>
```

Comment:
nit: I would rather add this as lib usage example (with a main function) in a lib-example directory(?) rather than as unit tests. What do you think?

Reply:
The idea is not to provide some usage examples, but to test somewhere the code that we put in the README, since that never gets compiled and run, and when we change something, those snippets can easily get out of date. 
Instead, in this way, if we modify something that breaks the README the tests fail. At least, this was the idea, I agree that nothing forces us to keep the test and the README snippets aligned, but at least, we knowingly have to fix at least the tests if we change something, so it cannot go completely unnoticed. Does it make sense to you?

Reply:
Yeah it does!